src/modules/fw_att_control/ecl_pitch_controller.cpp

```diff
@@ -102,11 +102,65 @@ float ECL_PitchController::control_bodyrate(const float dt, const ECL_ControlDat
 		_integrator = math::constrain(_integrator + id * _k_i, -_integrator_max, _integrator_max);
 	}
 
+	// Calculate the contribution of the rate gyro to the actuator demand and high pass filter
+	// The high pass filter reduces the contribution of the attitude control loop to the slew rate
+	// metric and reduces the likelihood that a loss of stability caused by a too small value of
+	// _tc would result in a an undesirable reduction in gyro rate feedback
+	// Note: high pass filter is implemented as HPF_output = input - LPF_output
+	const float p_term = _rate_error * _k_p * ctl_data.scaler * ctl_data.scaler;
+	float filter_input;
+	if (_k_ff > 0.0f) {
+		// If feed forward is being used then we can monitor the rate error term
+		// because the feed forward should be doing most of the work
+		filter_input = p_term;
+	} else {
+		// Some users will use a zero feed forward gain and rely on the rate error term
+		// to do all the work, resulting in large rate error tranisents. In this scenario
+		// it is better to monitor the gyro feedback contribution only
+		filter_input = ctl_data.body_y_rate * _k_p * ctl_data.scaler * ctl_data.scaler;
+	}
+	const float filt_tconst = _tc / 2.0f;
+	const float filt_coef = fminf(dt, filt_tconst) / filt_tconst;
+	_gyro_contribution_lpf = (1.0f - filt_coef) * _gyro_contribution_lpf + filt_coef * filter_input;
```

Comment:
You could use the AlphaFilter class we have in ECL for this:
https://github.com/PX4/PX4-Autopilot/blob/a11d2207e454e1a538e30aaccdfb4418b3e8accb/src/modules/mc_att_control/mc_att_control.hpp#L57

Reply:
OK will do.